operator/builtin/parser/csv/csv_test.go

```diff
@@ -0,0 +1,305 @@
+			map[string]interface{}{
+				"timestamp":    "20210316 17:08:01",
+				"serverhost":   "oiq-int-mysql",
+				"username":     "load",
+				"host":         "oiq-int-mysql.bluemedora.localnet",
+				"connectionid": "5",
+				"queryid":      "0",
+				"operation":    "DISCONNECT",
+				"database":     "",
+				"object":       "",
+				"retcode":      "0",
+			},
+		},
+		{
+			"empty field",
+			func(p *CSVParserConfig) {
+				p.Header = "name,address,age,phone,position"
+			},
+			"stanza,Evergreen,,555-5555,agent",
+			map[string]interface{}{
+				"name":     "stanza",
+				"address":  "Evergreen",
+				"age":      "",
+				"phone":    "555-5555",
+				"position": "agent",
+			},
+		},
+		{
+			"tab delimiter",
+			func(p *CSVParserConfig) {
+				p.Header = "name	address	age	phone	position"
+				p.FieldDelimiter = "\t"
+			},
+			"stanza	Evergreen	1	555-5555	agent",
+			map[string]interface{}{
+				"name":     "stanza",
+				"address":  "Evergreen",
+				"age":      "1",
+				"phone":    "555-5555",
+				"position": "agent",
+			},
+		},
+		{
+			"comma in quotes",
+			func(p *CSVParserConfig) {
+				p.Header = "name,address,age,phone,position"
+			},
+			"stanza,\"Evergreen,49508\",1,555-5555,agent",
+			map[string]interface{}{
+				"name":     "stanza",
+				"address":  "Evergreen,49508",
+				"age":      "1",
+				"phone":    "555-5555",
+				"position": "agent",
+			},
+		},
+		{
+			"quotes in quotes",
+			func(p *CSVParserConfig) {
+				p.Header = "name,address,age,phone,position"
+			},
+			"\"bob \"\"the man\"\"\",Evergreen,1,555-5555,agent",
+			map[string]interface{}{
+				"name":     "bob \"the man\"",
+				"address":  "Evergreen",
+				"age":      "1",
+				"phone":    "555-5555",
+				"position": "agent",
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := NewCSVParserConfig("test")
+			cfg.OutputIDs = []string{"fake"}
+			tc.configure(cfg)
+
+			ops, err := cfg.Build(testutil.NewBuildContext(t))
+			require.NoError(t, err)
+			op := ops[0]
+
+			fake := testutil.NewFakeOutput(t)
+			op.SetOutputs([]operator.Operator{fake})
+
+			entry := entry.New()
+			entry.Body = tc.inputBody
+			err = op.Process(context.Background(), entry)
+			require.NoError(t, err)
+
+			fake.ExpectBody(t, tc.outputBody)
+		})
+	}
+}
+
+func TestParserCSVMultipleBodys(t *testing.T) {
+	t.Run("basic", func(t *testing.T) {
+		cfg := NewCSVParserConfig("test")
+		cfg.OutputIDs = []string{"fake"}
+		cfg.Header = testHeader
+
+		ops, err := cfg.Build(testutil.NewBuildContext(t))
+		require.NoError(t, err)
+		op := ops[0]
+
+		fake := testutil.NewFakeOutput(t)
+		op.SetOutputs([]operator.Operator{fake})
+
+		entry := entry.New()
+		entry.Body = "stanza,INFO,started agent\nstanza,DEBUG,started agent"
+		err = op.Process(context.Background(), entry)
+		// require.Nil(t, err, "Expected to parse a single csv record, got '2'")
+		// require.Contains(t, err.Error(), "Expected to parse a single csv record, got '2'")
```

Comment:
We should understand why this is commented. Remove if not needed, or resolve the issue.